### PR TITLE
xcm fee of native token to treasury

### LINKED ISSUE #876: XCM fee of native asset go to Treasury
Current when receive KMA from other parachain, the xcm fee just burned, we can send it to treasury.

https://github.com/Manta-Network/Manta/blob/bfa08d62c5e93d4abd65c3d844494135b745d8d6/runtime/calamari/src/xcm_config.rs#L224

### PR DESCRIPTION
**Pull request by zqhxuyuan**
Signed-off-by: zqhxuyuan <zqhxuyuan@gmail.com>

## Description

closes: #876

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [x] Connected to an issue with discussion and accepted design using zenhub "Connect issue" button below
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari`, `A-dolphin` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

**Comment by Dengjianping** [comment]
Just an idea, should we give some xcm fee share to block producer, just like what we did for transaction fee?
But if  xcm activities is not that active, it's not worth to do it.

File: runtime/calamari/src/xcm_config.rs [hunk 1]
```diff
@@ -180,6 +181,24 @@ parameter_types! {
     pub XcmFeesAccount: AccountId = Treasury::account_id();
 }
 
+/// Xcm fee of native token
+pub struct XcmNativeFeeToTreasury;
```
Comment (Dengjianping) [inline runtime/calamari/src/xcm_config.rs:185]: Seems `XcmNativeFeeToTreasury` is not added to manta, not sure why.

File: runtime/calamari/tests/xcm_mock/xcm_tests.rs [hunk 2]
```diff
@@ -350,6 +352,90 @@ fn send_para_a_native_asset_to_para_b() {
...
+        assert_ok!(parachain::XTokens::transfer(
+            parachain::Origin::signed(ALICE),
+            parachain::CurrencyId::MantaCurrency(b_asset_id_on_a),
+            amount,
+            Box::new(VersionedMultiLocation::V1(dest)),
+            ADVERTISED_DEST_WEIGHT
+        ));
+        assert_eq!(
+            parachain::Assets::balance(b_asset_id_on_a, &ALICE),
+            INITIAL_BALANCE - amount
```
Comment (ferrell-code) [inline runtime/calamari/tests/xcm_mock/xcm_tests.rs]: This is just `INITIAL_BALANCE` -`INITIAL_BALANCE`, maybe just make zero?

Or perhaps the above line `let amount = INITIAL_BALANCE` should be a number like in the other tests

**Review by ferrell-code — COMMENTED** [review]
Looks reasonable to me

**Review by ghzlatarev — DISMISSED** [review]

File: runtime/calamari/src/xcm_config.rs (hunk 1 above)
Comment (zqhxuyuan) [reply runtime/calamari/src/xcm_config.rs:185]: Manta don't have treasury account yet. We can have a future PR for manta.

File: runtime/calamari/tests/xcm_mock/xcm_tests.rs (hunk 2 above)
Comment (zqhxuyuan) [reply runtime/calamari/tests/xcm_mock/xcm_tests.rs]: change to value 0, and in `send_para_b_asset_to_para_b_with_trader_and_fee` testcase, set amount to other value.

**Review by Dengjianping — APPROVED** [review]

**Review by ferrell-code — APPROVED** [review]

**Review by Garandor — APPROVED** [review]

**Pull request merged**